src/Compilers/CSharp/Portable/Compiler/Instrumentation.cs

```diff
@@ -199,11 +199,52 @@ public override BoundNode Visit(BoundNode node)
                         }
                         break;
                     case BoundKind.LocalDeclaration:
-                        if (statement.Syntax.Parent.Kind() == SyntaxKind.VariableDeclaration && ((VariableDeclarationSyntax)statement.Syntax.Parent).Variables.Count > 1)
+                        if (statement.Syntax.Parent.Kind() == SyntaxKind.VariableDeclaration)
                         {
-                            // This declaration is part of a MultipleLocalDeclarations statement,
-                            // and so should not be treated as a separate statement.
-                            return visited;
+                            VariableDeclarationSyntax declarationSyntax = (VariableDeclarationSyntax)statement.Syntax.Parent;
+                            if (declarationSyntax.Variables.Count > 1)
+                            {
+                                // This declaration is part of a MultipleLocalDeclarations statement,
+                                // and so should not be treated as a separate statement.
+                                return visited;
+                            }
+
+                            // A statement that represents the declarations in a using or fixed statement should not be treated as a separate statement.
+                            switch (declarationSyntax.Parent.Kind())
+                            {
+                                case SyntaxKind.UsingStatement:
```

Comment:
These are not the only valid parents. Local declarations can occur in many contexts.